### ISSUE TITLE
Character vector can not be created with 'char'

### ISSUE THREAD

**Issue by fplaza** [RcppCore/Rcpp#343]
It seems to me that CharacterVector and StringVector have the same behaviour. (correct me if I am wrong)
It is OK because as far as I know R treats single chars as strings.

However, creating a CharacterVector from char produce wrong results in Rcpp
Here is an example:

```c++
#include <Rcpp.h>
using namespace Rcpp;


// [[Rcpp::export]]
NumericMatrix test_character_vector()
{
  NumericMatrix m(3,3);
  rownames(m) = CharacterVector::create('a','b','c');
  return m;
}


/*** R
m1=matrix(nrow = 3,ncol=3)
rownames(m1)=c('a','b','c')
rownames(m1) #OK
  
m2=test_character_vector()
rownames(m2) # WRONG
*/
```

Result
```
> m1=matrix(nrow = 3,ncol=3)

> rownames(m1)=c('a','b','c')

> rownames(m1) #OK
[1] "a" "b" "c"

> m2=test_character_vector()

> rownames(m2) # WRONG
[1] "abc" "bc"  "c"  
```

**Comment by fplaza** [comment]
This is fixed by this commit bbe2bbef9ddecf30a2be526c166ea4e54ff21727

**Comment by eddelbuettel** [comment]
Pathch looks good per se but I am a little fearful about side effects.  Then again it is inside `Rcpp::internal`.   Any thoughts @kevinushey @jjallaire ?
